deepchecks/tabular/context.py

```diff
@@ -203,6 +204,39 @@ def __init__(
         if test and not train:
             raise DatasetValidationError('Can\'t initialize context with only test. if you have single dataset, '
                                          'initialize it as train')
+
+        # Handle model_classes
+        if model_classes and len(model_classes) == 0:
+            raise DeepchecksValueError('Received empty model_classes')
+        supported_models_link = doclink(
+            'supported-prediction-format',
+            template='For more information please refer to the Supported Models guide {link}')
+        if model_classes and sorted(model_classes) != model_classes:
+            raise DeepchecksValueError(f'Received unsorted model_classes. {supported_models_link}')
+
+        self._task_type, self._observed_classes, self._model_classes = infer_task_type_and_classes(
+            model, train, test, model_classes)
+
+        # Validate probas shape, and make sure we know about all classes if only probabilities have been passed
+        if y_proba_train is not None and not isinstance(y_proba_train, (Sequence, np.ndarray)):
+            raise ValueError('y_proba_train must be a matrix with n_samples rows and n_classes columns')
+        if y_proba_test is not None and not isinstance(y_proba_test, (Sequence, np.ndarray)):
+            raise ValueError('y_proba_test must be a matrix with n_samples rows and n_classes columns')
+        if y_proba_test is not None and np.array(y_proba_test).shape[1] != np.array(y_proba_train).shape[1]:
+            raise DeepchecksValueError(
+                f'y_proba_test with {len(y_proba_test[0])} columns does not match y_proba_train with'
+                f' {len(y_proba_train[0])} columns.'
+            )
+        proba_shape = len(y_proba_train[0]) if y_proba_train is not None else None
```

Comment:
Why not proba.shape[1]?

Reply:
I don't know that it's a numpy array / pandas. Could be list of lists and the code will still run.